pkg/util/docker/host_ip_provider.go

```diff
@@ -42,7 +42,7 @@ type hostIPProvider struct {
 
 func getDockerHostIPsUncached() []string {
 	var isHostMode bool
-	if mode, err := GetAgentContainerNetworkMode(); err != nil && mode == "host" {
+	if mode, err := GetAgentContainerNetworkMode(); err == nil && mode == containers.HostNetworkMode {
```

Comment:
Are we missing test coverage here ?

Reply:
we are. unfortunately is is tricky to test this as it depends on runtime environment.